metricbeat-tests/README.md

```diff
@@ -0,0 +1,203 @@
+
+    // network alias to enable network discovery: metricbeat will use this as the service to monitor
+	redisService.SetNetworkAlias("redis_" + redisVersion + "-metricbeat_" + metricbeatVersion)
+
+    // runs the service to mmonitor
+	return serviceManager.Run(redisService)
+}
+
+// Matches the 'And metricbeat "<metricbeat_version>" is installed and configured for Redis module'
+// There are one input parameter
+// The function must return error, so that the test fails if metricbeat configured for the
+// redis service is not run
+func metricbeatIsInstalledAndConfiguredForRedisModule(metricbeatVersion string) error {
+	s, err := RunMetricbeatService(metricbeatVersion, redisService)
+
+    // this global variable is defined at test runner level
+    // we populate it here so that it can be used in reusable steps/functions
+	metricbeatService = s
+
+    // each module must contribute some specific fields for querying elasticsearch
+	query = ElasticsearchQuery{
+		EventModule:    "redis",                     // event.module field
+		ServiceVersion: redisService.GetVersion(),   // service.version field
+	}
+
+	return err
+}
+
+// Here it happens the magic!
+// The test runner parses the feature file and assigns a feature step with a Golang function
+// But no worries, if you leave this file empty, and you run the tests, the test runner will
+// complain that no implemetation has been found, writing in console the signatures missing
+// (basically the above functions but empty, and exactly the below context suite)
+func RedisFeatureContext(s *godog.Suite) {
+	s.Step(`^Redis "([^"]*)" is running for metricbeat "([^"]*)"$`, redisIsRunningForMetricbeat)
+    s.Step(`^metricbeat "([^"]*)" is installed and configured for Redis module$`, metricbeatIsInstalledAndConfiguredForRedisModule)
+    // this last function is common a to all the modules
+    // so it will be declared at framework/test runner level (see runner_test.go)
+	s.Step(`^there are no errors in the "([^"]*)" index$`, thereAreNoErrorsInTheIndex)
+}
+
+```
+
+## Running the tests
+At this moment, the CLI and the functional tests coexist in the same repository, that's why we are building the CLI to get access to its features. Eventually that would change and we would consume it as a binary. Meanwhile:
+
+```shell
+$ make build-binary        # generates the binary from the repository
+$ make run-elastic-stack   # runs the stack for metricbeat
+$ make functional-tests    # runs the test suite
+```
+
+You could set up the environment so that it's possible to run one single module. As we are using _tags_ for matching modules, we could tell `make` to run just the tests for redis:
+
+```shell
+$ LOG_LEVEL=DEBUG FLAG="-t" FEATURE="redis" make functional-test
+```
+where:
+
+- LOG_LEVEL: sets the default log level in the tool (DEBUG, INFO, WARN, ERROR, FATAL)
+- FLAG: if set as `-t`, it will tell `Godog` to fileter by tag.
+- FEATURE: sets the tag to filter by (apache, mysql, redis)
+
```

Comment:
I'd also add how to test a feature using `godog` command.

Reply:
Added!